core/sql/optimizer/OptPhysRelExpr.cpp

```diff
@@ -11814,10 +11814,22 @@ void SortGroupBy::addArrangementAndOrderRequirements(
   {
     // Shouldn't/Can't add a sort order type requirement
     // if we are in DP2
-    if (rg.getStartRequirements()->executeInDP2())
-      rg.addArrangement(groupExpr(),NO_SOT);
-    else
-      rg.addArrangement(groupExpr(),ESP_SOT);
+
+   if( NOT extraOrderExpr().isEmpty())
+   {
+     ValueIdList groupExprCpy(groupExpr());
+     for (CollIndex i=0; i< extraOrderExpr().entries(); i++)
```

Comment:
Instead of this for loop, I think you can just do a single statement:
`groupExprCpy.insert(extraOrderExpr());`